File: 2775/2775.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    2775.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/09/27 16:06:13 by jihoolee          #+#    #+#              #
#    Updated: 2021/09/27 16:42:36 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_people_num(floor, room_num):
	apt = [list(range(1, room_num + 1))]
	for _ in range(1, floor + 1):
		curr_floor = []
		for n in range(1, room_num + 1):
			curr_floor.append(sum(apt[-1][0: n]))
		apt.append(curr_floor)
	return (apt[-1][-1])

result = []

for _ in range(int(input())):
	k = int(input())
	n = int(input())
	result.append(get_people_num(k, n))

for i in result:
	print(i)

File: 2775/2775.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   2775.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/09/27 15:22:57 by jihoolee          #+#    #+#             */
/*   Updated: 2021/09/27 16:42:25 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

using namespace std;

int	sum(vector<int> vec, int len)
{
	int	result = 0;

	for (int i = 0; i < len; i++)
		result += vec[i];
	return (result);
}

int	get_people_num(int floor, int room_num)
{
	vector<vector<int> >	apt(floor + 1, vector<int>(room_num));
	int						total = 0;

	for (int n = 1; n <= room_num; n++)
		apt[0][n - 1] = n;
	for (int f = 1; f < floor + 1; f++)
		for (int n = 1; n <= room_num; n++)
			apt[f][n - 1] = sum(apt[f - 1], n);
	return (apt[floor][room_num - 1]);
}

int	main(void)
{
	int			cases;
	cin >> cases;
	int			floor, room_num;
	vector<int>	result(cases);

	for (int i = 0; i < cases; i++)
	{
		cin >> floor >> room_num;
		result[i] = get_people_num(floor, room_num);
	}
	for (int i = 0; i < cases; i++)
		cout << result[i] << endl;
	return (0);
}
